public/app/features/transformers/calculateHeatmap/heatmap.ts

```diff
@@ -38,11 +38,12 @@ export const heatmapTransformer: SynchronousDataTransformerInfo<HeatmapTransform
   },
 };
 
-export function sortAscStrInf(aName?: string | null, bName?: string | null) {
-  let aBound = aName === '+Inf' ? Infinity : +(aName ?? 0);
-  let bBound = bName === '+Inf' ? Infinity : +(bName ?? 0);
+function parseNumeric(v?: string | null) {
+  return v === '+Inf' ? Infinity : v === '-Inf' ? -Infinity : +(v ?? 0);
+}
 
-  return aBound - bBound;
+export function sortAscStrInf(aName?: string | null, bName?: string | null) {
+  return parseNumeric(aName) - parseNumeric(bName);
```

Comment:
this is currently unused, since `prepBucketFrames()` is unused.

will do a follow-up at some point to use this instead of what's currently inside of Prometheus `transformV2()`